server/src/test/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStreamTests.java

```diff
@@ -232,4 +266,13 @@ public void close() throws IOException {
             super.close();
         }
     }
+
+    private static final class IncreasingBytesUnlimitedInputStream extends InputStream {
+        long currentByte = 0;
+
+        @Override
+        public int read() throws IOException {
+            return (int) (currentByte++ % 256);
```

Comment:
This doesn't reliably demonstrate the problem since 256 divides evenly into `1<<32` so an integer overflow doesn't change the content seen - we're relying entirely on checking that `markedSliceOffset` is negative, which only happens if `mark` is just over 6GiB (i.e. `randomIntBetween(5, 6)` returns `6`).

I'd suggest using 255 instead, and also maybe a little more randomness in the choice of `mark`.